ImageGallery/ImageGallery/ImageStore 2.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  DownloadStore.swift
+//  ImagaeGallery
+//
+//  Created by Nick Ryan on 6/18/24.
+//
+
+import SwiftUI
+import Combine
+import Foundation
+
+@MainActor
+class ImageStore: NSObject, ObservableObject, URLSessionDownloadDelegate {
+  @Published var photos: [PexelsPhoto] = []
+  
+  private var session: URLSession
+  private var progressHandler: ((Double) -> Void)?
+  private var downloadTask: URLSessionDownloadTask?
+  private var downloadCompletionHandler: ((URL) -> Void)?
+  
+  override init() {
+    let sessionConfiguration = URLSessionConfiguration.default
+    self.session = URLSession(configuration: sessionConfiguration, delegate: nil, delegateQueue: nil)
+    super.init()
+    self.session = URLSession(configuration: sessionConfiguration, delegate: self, delegateQueue: nil)
+  }
+  
+  private var apiKey: String {
+    guard let filePath = Bundle.main.path(forResource: "DownloadInfo", ofType: "plist") else {
+      fatalError("Couldn't find 'DownloadInfo.plist'.")
+    }
+    
+    let plist = NSDictionary(contentsOfFile: filePath)
+    guard let value = plist?.object(forKey: "ImageAPI") as? String else {
+      fatalError("Couln't find key 'ImageAPI' in DownloadInfo.plist'.")
+    }
+    return value
+  }
+  
+  
+  func fetchImages(query: String) async {
+    print("Start fetching images for query: \(query)")
+    var components = URLComponents(string: "https://api.pexels.com/v1/search")!
+    
+    let queryItems = [
+      URLQueryItem(name: "query", value: query),
+      URLQueryItem(name: "per_page", value: "15")
+    ]
+    
+    components.queryItems = queryItems
+    
+    guard let url = components.url else {
+      fatalError("Invalid URL")
```

Comment:
careful now. This will crash the app. Perhaps its best to throw an error and handle it in the view.